src/Assignments/ProblemIframe.tsx

```diff
@@ -137,21 +139,31 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
     }
 
     function insertListener(problemForm: HTMLFormElement) {
-        problemForm.addEventListener('submit', _.debounce((event: { preventDefault: () => void; }) => {
-            event.preventDefault();
+        const submitHandler = (clickedButton: HTMLButtonElement) => {
             if (_.isNil(problemForm)) {
                 console.error('Hijacker: Could not find the form when submitting the form');
                 setError('An error occurred');
                 return;
             }
-            let clickedButton = problemForm.querySelector('.btn-clicked') as HTMLButtonElement;
             if (_.isNil(clickedButton)) {
                 setError('Hijacker: An error occurred');
                 console.error('Could not find the button that submitted the form');
                 return;
             }
             prepareAndSubmit(problemForm, clickedButton);
-        }, 4000, {leading: true, trailing:false}));
+        };
+        const debouncedSubmitHandler = _.debounce(submitHandler, 4000, { leading: true, trailing: false });
```

Comment:
I wonder what happens if you set leading and trailing to true 🤔 

I would think it would run immediately and if you clicked again it would run after the delay, multiple clicks would dealy the dealy event